### ISSUE TITLE
similarity as alternative way of searching

### ISSUE THREAD

**Issue by dfa1** [hosh-shell/hosh#267]
**Problem**
Provide a command to search by similarity, according to different well known algorithms such as 
Levenshtein, Cosine, Jaccard.

**Solution**
In practice, create a tiny wrapper around this library:
https://github.com/tdebatty/java-string-similarity

Normalized values would require also a new value type (decimal or float).

**Benefit**
Being able to search not only by exact match or regexp but also by similarity.

**Comment by kosiakk** [comment]
Smart synonym search would also be magnificent. Some embedding like `Word2Vec` or more modern from GPT-2. I'd love play around with them.
Look, [lots of guys](https://huggingface.co/transformers/pretrained_models.html) play with them for fun, pre-training those enormous models on some specific area. Ask them to train a model on bash scripts and comments. It will then write code for you, from the comments :-)
